servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -3829,9 +3829,41 @@ public void deleteRole(ResourceContext ctx, String domainName, String roleName,
             throw ZMSUtils.requestError("deleteRole: admin role cannot be deleted", caller);
         }
 
+        /* we are not going to allow any user to delete
+         * role that related to any assertion of any policy */
+
+        List<Policy> policyList = getPolicyList(domainName, caller);
+
+        if (policyList != null) {
+            validateRoleNotAssociatedToPolicy(policyList, roleName, domainName, caller);
+        }
+
         dbService.executeDeleteRole(ctx, domainName, roleName, auditRef, caller);
     }
 
+    private List<Policy> getPolicyList(String domainName, String caller) {
+        Policies result = new Policies();
+        AthenzDomain domain = getAthenzDomain(domainName, false);
+        if (domain == null) {
+            throw ZMSUtils.notFoundError(caller + ": Domain not found: '" + domainName + "'", caller);
+        }
+        return result.setList(setupPolicyList(domain, true, true)).getList();
```

Comment:
why are creating a Policies object, do setList and then do getList()?

we should able just return setupPolicyList(domain, true, true) without creating the result object.